code/game/objects/items/storage/pouch.dm

```diff
@@ -0,0 +1,359 @@
+/obj/item/storage/pouch
+	name = "abstract pouch"
+	icon = 'icons/obj/clothing/pouches.dmi'
+	icon_state = "small_drop"
+	w_class = WEIGHT_CLASS_BULKY //does not fit in backpack
+	slot_flags = ITEM_SLOT_POCKET
+	var/fill_type
+	var/fill_number = 0
+
+/obj/item/storage/pouch/Initialize()
+	. = ..()
+	if(fill_number && fill_type)
+		for(var/i in 1 to fill_number)
+			new fill_type(src)

```

Comment:
does this actually work

Reply:
not really sure, more or less just lifted it from TGMC and changed the stuff that needed to be changed to get it minimally working.

Reply:
to clarify, the pouches work fine, not sure if this particular code actually does anything though

Reply:
I doubt it, you're putting them inside of the pouch, theoretically making sense but we store items in the component, not the contents. (afaik)

Reply:
also tgmc != TG so do not use them as example on most things.

Reply:
i mean, i can just delete that part if you want, i didnt even really understand what purpose it served

Reply:
i have killed that code, as well as a bit of other stuff that i didnt understand the purpose of (the only place where the fill stuff was used was the medkit pouch, which i switched to a populatecontents function).  is this better?

Reply:
also fixed the examine function